flang/lib/Optimizer/Transforms/OMPMapInfoFinalization.cpp

```diff
@@ -59,6 +59,34 @@ class OMPMapInfoFinalizationPass
            /*corresponding local alloca=*/fir::AllocaOp>
       localBoxAllocas;
 
+  unsigned long getDescriptorMapType(unsigned long mapTypeFlag,
+                                     mlir::Operation *target) {
+    auto newDescFlag = llvm::omp::OpenMPOffloadMappingFlags(mapTypeFlag);
+
+    if ((llvm::isa_and_nonnull<mlir::omp::TargetDataOp>(target) ||
+         llvm::isa_and_nonnull<mlir::omp::TargetOp>(target)) &&
+        static_cast<
+            std::underlying_type_t<llvm::omp::OpenMPOffloadMappingFlags>>(
+            (newDescFlag &
+             llvm::omp::OpenMPOffloadMappingFlags::OMP_MAP_FROM)) &&
```

Comment:
Would something like this be equivalent? I'm just suggesting it because these `if` statements are quite difficult to read currently:
```suggestion
        llvm::to_underlying(newDescFlag & llvm::omp::OpenMPOffloadMappingFlags::OMP_MAP_FROM) &&
```